lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexDriver.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver;
+
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.reef.driver.context.ActiveContext;
+import org.apache.reef.driver.context.ContextConfiguration;
+import org.apache.reef.driver.context.ContextMessage;
+import org.apache.reef.driver.context.FailedContext;
+import org.apache.reef.driver.evaluator.*;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Configurations;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.annotations.Unit;
+import org.apache.reef.vortex.api.VortexThreadPool;
+import org.apache.reef.vortex.api.VortexUserCode;
+import org.apache.reef.vortex.common.ReportTaskletResult;
+import org.apache.reef.vortex.driver.tasklet.DefaultTaskletExecutor;
+import org.apache.reef.vortex.evaluator.ExecutorConfiguration;
+import org.apache.reef.vortex.evaluator.LocalExecutor;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * REEF Driver for Vortex.
+ */
+@Unit
+public final class VortexDriver {
+  private static final Logger LOG = Logger.getLogger(VortexDriver.class.getName());
+  private final EvaluatorRequestor evaluatorRequestor;
+  private final VortexManager vortexManager;
+  private final CommandSender commandSender;
+
+  private final int evalMem;
+  private final int evalNum;
```

Comment:
The initial evalNum is fixed. But the actual number of evaluators can change over time due to elasticity. We should be able to maintain this information. 
Do we target homogeneous evaluators? This is a good start. I'm just checking if we need to support heterogeneous evaluators for the first target Vortex application.